### ISSUE TITLE
Nan

### ISSUE THREAD

**Issue by 1559092344** [X-Square-Robot/wall-x#32]
Hi! I used the data from your demo.  NaN loss ​​displayed during training.  What could be the possible reason?

**Comment by 1559092344** [comment]
> Hi [@1559092344](https://github.com/1559092344) ,
> 
> Thanks for reporting this issue!
> 
> We haven't encountered NaN loss issues during our testing with the demo data. To help us investigate this problem, could you please provide:
> 
>     1. **Complete configuration file** - Your training config/settings
> 
>     2. **Training logs** - Full log output showing when the NaN loss occurs
> 
>     3. **Environment details** - Python version, framework versions, hardware specs
> 
> 
> Thanks for your patience!

Thx for your reply. My hardware is  A100, 80GB. The following are the relevant details: the training log (can find  config_qact.yml ), environment configuration information, and the dataset's info.json file. Notably, in config_qact.yml , I read the aloha_mobile_cabinet data from the local path (root),because it is hard to connect to huggingface during my training,. In load_lerobot_dataset.py , I changed "dataset = LeRobotDataset(repo_id, delta_timestamps=delta_timestamps, video_backend="pyav")" to "dataset = LeRobotDataset(repo_id, root=lerobot_config.get("root", None), delta_timestamps=delta_timestamps, video_backend="pyav")." . By the way,  I  see Selected episodes: None
Number of episodes selected: 85
Number of frames selected: 127500errors in the training log. I suspect this is due to the Lerobot version. Any feedback is welcome.


Running command: accelerate launch --num_processes=1 --main_process_port=21805 /data/algorithm/wall/train_qact.py --config /data/algorithm/wall/workspace/lerobot_example/config_qact.yml --seed 10239
[HAMI-core Msg(2078578:140101050135040:libvgpu.c:838)]: Initializing.....
The following values were not passed to `accelerate launch` and had defaults used instead:
        `--num_machines` was set to a value of `1`
        `--mixed_precision` was set to a value of `'no'`
        `--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
[HAMI-core Msg(2078642:140078056739328:libvgpu.c:838)]: Initializing.....
[HAMI-core Msg(2078642:140078056739328:libvgpu.c:857)]: Initialized
[2025-09-17 00:52:46] Preparing accelerator
[2025-09-17 00:52:46] Accelerator initialization complete
Configuration:
==================================================
{
  "log_name": "robotic_training",
  "log_project": "vla_training",
  "model_type": "qwen2_5",
  "pretrained_wallx_path": "/data/algorithm/wall-oss-flow",
  "save_path": ".",
  "use_fast_tokenizer": false,
  "action_tokenizer_path": "/path/to/fast/",
  "profile": false,
  "profile_save_path": "/path/to/profile/",
  "profile_wait_iters": 10,
  "profile_warmup_iters": 5,
  "profile_active_iters": 2,
  "num_warmup_steps": 100,
  "num_training_steps": 64000000,
  "learning_rate": 9e-05,
  "min_lr": 5e-05,
  "num_epoch": 100,
  "gradient_accumulation_steps": 32,
  "batch_size_per_gpu": 1,
  "padding_side": "left",
  "epoch_save_interval": 10,
  "dof_config": {
    "follow_left_ee_cartesian_pos": 3,
    "follow_left_ee_rotation": 3,
    "follow_left_gripper": 1,
    "follow_right_ee_cartesian_pos": 3,
    "follow_right_ee_rotation": 3,
    "follow_right_gripper": 1,
    "head_actions": 2,
    "height": 1,
    "car_pose": 3
  },
  "agent_pos_config": {
    "follow_left_ee_cartesian_pos": 3,
    "follow_left_ee_rotation": 3,
    "follow_left_gripper": 1,
    "follow_right_ee_cartesian_pos": 3,
    "follow_right_ee_rotation": 3,
    "follow_right_gripper": 1,
    "head_actions": 2,
    "height": 1,
    "car_pose": 3
  },
  "data": {
    "use_lerobot": true,
    "lerobot_config": {
      "repo_id": "lerobot/aloha_mobile_cabinet",
      "root": "/data/datasets/wallx-data/newdata/newdata",
      "episodes": null,
      "image_transforms": null,
      "delta_timestamps": null,
      "tolerance_s": "1e-4",
      "revision": null,
      "force_cache_sync": false,
      "download_videos": true,
      "video_backend": null
    },
"action_horizon": 32,
    "train_test_split": 0.95,
    "obs_action_keys": [
      "follow_left_ee_cartesian_pos",
      "follow_left_ee_rotation",
      "follow_left_gripper",
      "follow_right_ee_cartesian_pos",
      "follow_right_ee_rotation",
      "follow_right_gripper",
      "head_actions",
      "height",
      "car_pose"
    ],
    "predict_action_keys": [
      "follow_left_ee_cartesian_pos",
      "follow_left_ee_rotation",
      "follow_left_gripper",
      "follow_right_ee_cartesian_pos",
      "follow_right_ee_rotation",
      "follow_right_gripper",
      "head_actions",
      "height",
      "car_pose"
    ],
    "resolution": {
      "face_view": 256,
      "left_wrist_view": 256,
      "right_wrist_view": 256,
      "move1_view": 256,
      "move2_view": 256,
      "top_view": 256,
      "wall_view": 256,
      "multi_modal": 256
    },
    "model_type": "qwen2_5"
  }
}
==================================================
wandb: (1) Create a W&B account
wandb: (2) Use an existing W&B account
wandb: (3) Don't visualize my results
wandb: Enter your choice: 3
wandb: You chose "Don't visualize my results"
wandb: Tracking run with wandb version 0.21.4
wandb: W&B syncing is set to `offline` in this directory. Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
wandb: Run data is saved locally in /data/algorithm/wall/workspace/lerobot_example/wandb/offline-run-20250917_005250-8ri1l5l5
You are attempting to use Flash Attention 2.0 without specifying a torch dtype. This might lead to unexpected behaviour
ActionProcessor Configuration:
  Action dimension: 20
  Proprioception dimension: 20
  DOF configuration:
    follow_left_ee_cartesian_pos: 3
    follow_left_ee_rotation: 3
    follow_left_gripper: 1
    follow_right_ee_cartesian_pos: 3
    follow_right_ee_rotation: 3
    follow_right_gripper: 1
    head_actions: 2
    height: 1
    car_pose: 3
  Agent position configuration:
    follow_left_ee_cartesian_pos: 3
    follow_left_ee_rotation: 3
    follow_left_gripper: 1
    follow_right_ee_cartesian_pos: 3
    follow_right_ee_rotation: 3
    follow_right_gripper: 1
    head_actions: 2
    height: 1
    car_pose: 3
The new embeddings will be initialized from a multivariate normal distribution that has old embeddings' mean and covariance. As described in this article: https://nlp.stanford.edu/~johnhew/vocab-expansion.html. To disable this, use `mean_resizing=False`
[current time: 2025-09-17 00:54:56] Function load_model took 117.20 seconds to execute
rank 0 after load model memory usage: 7.98 GB
Loading Vision-Language-Action data from /data/algorithm/wall/wall_x/trainer/qwen_vl_act_trainer.py
Resolving data files: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 85/85 [00:00<00:00, 84562.58it/s]
Selected episodes: None
Number of episodes selected: 85
Number of frames selected: 127500

=============================
LeRobot Data Loading Configuration:
✦ RANK: 0
✦ WORLD SIZE: 1
✦ BATCH SIZE PER GPU: 1
✦ REPO ID: lerobot/aloha_mobile_cabinet
✦ TOTAL DATASET SIZE: 127500
✦ TOTAL BATCHES: 127500
✦ SEED: 42
==================================================

[current time: 2025-09-17 00:55:03] Function load_qact_data took 7.15 seconds to execute
rank 0 after load qact data usage: 7.98 GB
[current time: 2025-09-17 00:55:03] Function __init__ took 124.39 seconds to execute
/data/algorithm/wall/wall_x/fusions/ops.py:438: UserWarning: The input `q` of multimodal_rope op is discontiguous!
  warnings.warn("The input `q` of multimodal_rope op is discontiguous!")
/data/algorithm/wall/wall_x/fusions/ops.py:441: UserWarning: The input `k` of multimodal_rope op is discontiguous!
  warnings.warn("The input `k` of multimodal_rope op is discontiguous!")
/data/algorithm/wall/wall_x/fusions/ops.py:220: UserWarning: The data type of the input `indices` of permute_topK op is torch.bool! The recommended type is torch.int32.
  warnings.warn(
Warning: NaN loss detected in epoch: 0, step: 0
Warning: NaN loss detected in epoch: 0, step: 1
Warning: NaN loss detected in epoch: 0, step: 2
Warning: NaN loss detected in epoch: 0, step: 3
Warning: NaN loss detected in epoch: 0, step: 4
Warning: NaN loss detected in epoch: 0, step: 5
Warning: NaN loss detected in epoch: 0, step: 6
Warning: NaN loss detected in epoch: 0, step: 7
Warning: NaN loss detected in epoch: 0, step: 8
Warning: NaN loss detected in epoch: 0, step: 9
Warning: NaN loss detected in epoch: 0, step: 10
Warning: NaN loss detected in epoch: 0, step: 11
Warning: NaN loss detected in epoch: 0, step: 12
Warning: NaN loss detected in epoch: 0, step: 13
Warning: NaN loss detected in epoch: 0, step: 14
Warning: NaN loss detected in epoch: 0, step: 15
Warning: NaN loss detected in epoch: 0, step: 16
Warning: NaN loss detected in epoch: 0, step: 17





Package                  Version     Editable project location
------------------------ ----------- ----------------------------
accelerate               1.10.1
aiohappyeyeballs         2.6.1
aiohttp                  3.12.15
aiosignal                1.4.0
annotated-types          0.7.0
async-timeout            5.0.1
attrs                    25.3.0
av                       15.1.0
blinker                  1.9.0
certifi                  2025.8.3
charset-normalizer       3.4.3
click                    8.2.2
cloudpickle              3.1.1
cmake                    4.1.0
datasets                 3.6.0
deepdiff                 8.6.1
diffusers                0.35.1
dill                     0.3.8
draccus                  0.10.0
einops                   0.8.1
evdev                    1.9.2
Farama-Notifications     0.0.4
filelock                 3.19.1
flash_attn               2.7.4.post1
Flask                    3.1.2
frozenlist               1.7.0
fsspec                   2025.3.0
gitdb                    4.0.12
GitPython                3.1.45
gymnasium                0.29.1
hf_transfer              0.1.9
hf-xet                   1.1.10
huggingface-hub          0.34.4
idna                     3.10
imageio                  2.37.0
imageio-ffmpeg           0.6.0
importlib_metadata       8.7.0
inquirerpy               0.3.4
itsdangerous             2.2.0
Jinja2                   3.1.6
jsonlines                4.0.0
lerobot                  0.3.4       /data/algorithm/lerobot-main
MarkupSafe               3.0.2
mergedeep                1.3.4
mpmath                   1.3.0
multidict                6.6.4
multiprocess             0.70.16
mypy_extensions          1.1.0
networkx                 3.4.2
numpy                    2.2.6
nvidia-cublas-cu12       12.4.5.8
nvidia-cuda-cupti-cu12   12.4.127
nvidia-cuda-nvrtc-cu12   12.4.127
nvidia-cuda-runtime-cu12 12.4.127
nvidia-cudnn-cu12        9.1.0.70
nvidia-cufft-cu12        11.2.1.3
nvidia-curand-cu12       10.3.5.147
nvidia-cusolver-cu12     11.6.1.9
nvidia-cusparse-cu12     12.3.1.170
nvidia-cusparselt-cu12   0.6.2
nvidia-nccl-cu12         2.21.5
nvidia-nvjitlink-cu12    12.4.127
nvidia-nvtx-cu12         12.4.127
opencv-python-headless   4.12.0.88
orderly-set              5.5.0
packaging                25.0
pandas                   2.3.2
peft                     0.17.1
pfzy                     0.3.4
pillow                   11.3.0
pip                      25.2
platformdirs             4.4.0
prompt_toolkit           3.0.52
propcache                0.3.2
protobuf                 6.32.1
psutil                   7.0.0
pyarrow                  21.0.0
pydantic                 2.11.9
pydantic_core            2.33.2
pynput                   1.8.1
pyserial                 3.5
python-dateutil          2.9.0.post0
python-xlib              0.33
pytz                     2025.2
PyYAML                   6.0.2
pyyaml-include           1.4.1
qwen-vl-utils            0.0.11
regex                    2025.9.1
requests                 2.32.5
rerun-sdk                0.22.1
safetensors              0.6.2
scipy                    1.15.3
sentry-sdk               2.37.1
setuptools               78.1.1
six                      1.17.0
smmap                    5.0.2
sympy                    1.13.1
termcolor                3.1.0
tokenizers               0.21.4
toml                     0.10.2
torch                    2.6.0
torchaudio               2.6.0
torchcodec               0.5
torchdiffeq              0.2.5
torchvision              0.21.0
tqdm                     4.67.1
transformers             4.49.0
triton                   3.2.0
typing_extensions        4.15.0
typing-inspect           0.9.0
typing-inspection        0.4.1
tzdata                   2025.2
urllib3                  2.5.0
wall_x                   1.0.0
wandb                    0.21.4
wcwidth                  0.2.13
Werkzeug                 3.1.3
wheel                    0.45.1
xxhash                   3.5.0
yarl                     1.20.1
zipp                     3.23.0





{
    "codebase_version": "v2.1",
    "robot_type": "aloha",
    "total_episodes": 85,
    "total_frames": 127500,
    "total_tasks": 1,
    "total_videos": 255,
    "total_chunks": 1,
    "chunks_size": 1000,
    "fps": 50,
    "splits": {
        "train": "0:85"
    },
    "data_path": "data/chunk-{episode_chunk:03d}/episode_{episode_index:06d}.parquet",
    "video_path": "videos/chunk-{episode_chunk:03d}/{video_key}/episode_{episode_index:06d}.mp4",
    "features": {
        "observation.images.cam_high": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "names": [
                "height",
                "width",
                "channel"
            ],
            "video_info": {
                "video.fps": 50.0,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "has_audio": false
            }
        },
        "observation.images.cam_left_wrist": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "names": [
                "height",
                "width",
                "channel"
            ],
            "video_info": {
                "video.fps": 50.0,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "has_audio": false
            }
        },
        "observation.images.cam_right_wrist": {
            "dtype": "video",
            "shape": [
                480,
                640,
                3
            ],
            "names": [
                "height",
                "width",
                "channel"
            ],
            "video_info": {
                "video.fps": 50.0,
                "video.codec": "av1",
                "video.pix_fmt": "yuv420p",
                "video.is_depth_map": false,
                "has_audio": false
            }
        },
        "observation.state": {
            "dtype": "float32",
            "shape": [
                14
            ],
            "names": {
                "motors": [
                    "left_waist",
                    "left_shoulder",
                    "left_elbow",
                    "left_forearm_roll",
                    "left_wrist_angle",
                    "left_wrist_rotate",
                    "left_gripper",
                    "right_waist",
                    "right_shoulder",
                    "right_elbow",
                    "right_forearm_roll",
                    "right_wrist_angle",
                    "right_wrist_rotate",
                    "right_gripper"
                ]
            }
        },
        "observation.effort": {
            "dtype": "float32",
            "shape": [
                14
            ],
            "names": {
                "motors": [
                    "left_waist",
                    "left_shoulder",
                    "left_elbow",
                    "left_forearm_roll",
                    "left_wrist_angle",
                    "left_wrist_rotate",
                    "left_gripper",
                    "right_waist",
                    "right_shoulder",
                    "right_elbow",
                    "right_forearm_roll",
                    "right_wrist_angle",
                    "right_wrist_rotate",
                    "right_gripper"
                ]
            }
        },
        "action": {
            "dtype": "float32",
            "shape": [
                14
            ],
            "names": {
                "motors": [
                    "left_waist",
                    "left_shoulder",
                    "left_elbow",
                    "left_forearm_roll",
                    "left_wrist_angle",
                    "left_wrist_rotate",
                    "left_gripper",
                    "right_waist",
                    "right_shoulder",
                    "right_elbow",
                    "right_forearm_roll",
                    "right_wrist_angle",
                    "right_wrist_rotate",
                    "right_gripper"
                ]
            }
        },
        "episode_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "frame_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "timestamp": {
            "dtype": "float32",
            "shape": [
                1
            ],
            "names": null
        },
        "next.done": {
            "dtype": "bool",
            "shape": [
                1
            ],
            "names": null
        },
        "index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        },
        "task_index": {
            "dtype": "int64",
            "shape": [
                1
            ],
            "names": null
        }
    }
}
